### ISSUE TITLE
Segfault when closing rock-display with osgviz

### ISSUE THREAD

**Issue by chhtz** [rock-gui/gui-osgviz-osgviz#6]
When I start rock-display and visualize anything using osgviz (e.g., RigidBodyState visualization) everything works fine, until I close rock-display, after which I get a segfault with the following error output (I removed the majority of the memory map section). Essentially the same is true for any ruby script which uses 3D-visualization of any kind.

This is quite annoying especially if one wants to read the last (regular) output lines of a script ...

```
chtz@chertzberg-u:~/workspace/entern$ rock-display 
Bundles[INFO]: Active bundles: entern, rock
orocos.rb[WARN]: /taskManagement is a task context not generated by orogen, using default task model
[BUG] Segmentation fault
ruby 2.0.0p384 (2014-01-12) [x86_64-linux-gnu]

-- C level backtrace information -------------------------------------------
/usr/lib/x86_64-linux-gnu/libruby-2.0.so.2.0(+0x177bbb) [0x7f0c40f45bbb]
/usr/lib/x86_64-linux-gnu/libruby-2.0.so.2.0(+0x64afa) [0x7f0c40e32afa] vfscanf.c:1394
/usr/lib/x86_64-linux-gnu/libruby-2.0.so.2.0(rb_bug+0xb3) [0x7f0c40e33203] vfscanf.c:858
/usr/lib/x86_64-linux-gnu/libruby-2.0.so.2.0(+0x112196) [0x7f0c40ee0196] getnameinfo.c:153
/lib/x86_64-linux-gnu/libc.so.6(+0x36cb0) [0x7f0c40a3fcb0] ../sysdeps/posix/killpg.c:37
/usr/lib/libosgText.so.99(_ZNK7osgText4Text16releaseGLObjectsEPN3osg5StateE+0x19) [0x7f0becb64c29] strtod_l.c:351
/usr/lib/libosg.so.99(_ZNK3osg5Geode16releaseGLObjectsEPNS_5StateE+0x30) [0x7f0bed425e10]
/usr/lib/libosg.so.99(_ZNK3osg5Group16releaseGLObjectsEPNS_5StateE+0x30) [0x7f0bed44fa00]
/usr/lib/libosg.so.99(_ZNK3osg5Group16releaseGLObjectsEPNS_5StateE+0x30) [0x7f0bed44fa00]
/usr/lib/libosg.so.99(_ZNK3osg5Group16releaseGLObjectsEPNS_5StateE+0x30) [0x7f0bed44fa00]
/usr/lib/libosg.so.99(_ZNK3osg5Group16releaseGLObjectsEPNS_5StateE+0x30) [0x7f0bed44fa00]
/usr/lib/libosg.so.99(_ZNK3osg5Group16releaseGLObjectsEPNS_5StateE+0x30) [0x7f0bed44fa00]
/usr/lib/libosg.so.99(_ZNK3osg5Group16releaseGLObjectsEPNS_5StateE+0x30) [0x7f0bed44fa00]
/usr/lib/libosg.so.99(_ZNK3osg5Group16releaseGLObjectsEPNS_5StateE+0x30) [0x7f0bed44fa00]
/usr/lib/libosg.so.99(_ZN3osg15GraphicsContext12removeCameraEPNS_6CameraE+0x1ce) [0x7f0bed44778e]
/usr/lib/libosg.so.99(_ZN3osg6CameraD1Ev+0x3f) [0x7f0bed3de00f]
/usr/lib/libosg.so.99(_ZN3osg6CameraD0Ev+0x9) [0x7f0bed3de5c9]
/usr/lib/libosg.so.99(_ZN3osg4ViewD2Ev+0x3b7) [0x7f0bed524c27]
/home/chtz/workspace/entern/install/lib/libosgViz.so(_ZN6osgviz9SuperViewD1Ev+0x1a8) [0x7f0bec411588] ../iconv/loop.c:448
/home/chtz/workspace/entern/install/lib/libosgViz.so(_ZN6osgviz9SuperViewD0Ev+0x9) [0x7f0bec4115d9] ../iconv/loop.c:437
/usr/lib/libosgViewer.so.99(_ZNSt6vectorIN3osg7ref_ptrIN9osgViewer4ViewEEESaIS4_EED1Ev+0x50) [0x7f0bec8ac1b0] vfscanf.c:1700
/usr/lib/libosgViewer.so.99(_ZN9osgViewer15CompositeViewerD2Ev+0x1ea) [0x7f0bec8a7f8a] vfscanf.c:1854
/home/chtz/workspace/entern/install/lib/libosgViz.so(_ZN6osgviz6WindowD1Ev+0x1df) [0x7f0bec40e0ff] gconv_conf.c:569
/home/chtz/workspace/entern/install/lib/libosgViz.so(_ZN6osgviz6WindowD0Ev+0x9) [0x7f0bec40e149] gconv_conf.c:384
/home/chtz/workspace/entern/install/lib/libosgViz.so(_ZN6osgviz13WindowManagerD1Ev+0x53) [0x7f0bec410a93] ../iconv/loop.c:437
/home/chtz/workspace/entern/install/lib/libosgViz.so(_ZN6osgviz13WindowManagerD0Ev+0x9) [0x7f0bec410ad9] ../iconv/loop.c:476
/home/chtz/workspace/entern/install/lib/libvizkit3d.so(_ZNK3osg10Referenced5unrefEv+0x1f) [0x7f0bed8a1daf] ../stdlib/gmp.h:459
/home/chtz/workspace/entern/install/lib/libosgViz.so(_ZN6osgviz6OsgVizD1Ev+0x85) [0x7f0bec40bef5] gconv_open.c:238
/home/chtz/workspace/entern/install/lib/libosgViz.so(_ZN6osgviz6OsgVizD0Ev+0x9) [0x7f0bec40bf59] gconv_open.c:265
/home/chtz/workspace/entern/install/lib/libosgViz.so(_ZN3osg7ref_ptrIN6osgviz6OsgVizEED1Ev+0x36) [0x7f0bec40c146] gconv_open.c:330
/lib/x86_64-linux-gnu/libc.so.6(+0x3c1a9) [0x7f0c40a451a9] exit.c:82
/lib/x86_64-linux-gnu/libc.so.6(+0x3c1f5) [0x7f0c40a451f5] exit.c:104
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xfc) [0x7f0c40a2af4c] libc-start.c:321
/usr/bin/ruby2.0() [0x400899]

-- Other runtime information -----------------------------------------------

* Process memory map:

00400000-00401000 r-xp 00000000 08:01 4735374                            /usr/bin/ruby2.0
00600000-00601000 r--p 00000000 08:01 4735374                            /usr/bin/ruby2.0
00601000-00602000 rw-p 00001000 08:01 4735374                            /usr/bin/ruby2.0
0069a000-083ea000 rw-p 00000000 00:00 0                                  [heap]

    2000+ Lines of Process memory map ...

7fff393fe000-7fff39400000 r-xp 00000000 00:00 0                          [vdso]
ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]


[NOTE]
You may have encountered a bug in the Ruby interpreter or extension libraries.
Bug reports are welcome.
For details: http://www.ruby-lang.org/bugreport.html
```

**Comment by arneboe** [comment]
Fixed by #8

**Comment by arneboe** [comment]
@planthaber please close
